Задание 1 - этап 2

seal-developer opened the pull request:
Добавила возможность удаления/добавления задачи.

seal-developer committed:
Задание 1, этап 2 - добавила возможность удаления/добавления задачи

seal-developer committed:
Изменила место исполнения сортировки

Tauders commented on ReactTasks/task-1/stage-2/.prettierrc.json:
```diff
@@ -0,0 +1,6 @@
+  "arrowParens": "avoid",
+  "endOfLine": "lf",
+  "tabWidth": 2
+}
```
Не хватает пустой строки в конце файла

seal-developer replied:
Добавила строку

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,31 @@
+}: ButtonProps) => {
+  let classesNames = className
+    ? className + ' ' + classes.button
+    : classes.button;
```
Здесь можно использовать библиотеку `classnames`, чтобы упростить условие

seal-developer replied:
Использовала библиотеку

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,31 @@
+  onClick,
+  onKeyDown,
+  children,
+  disabled,
```
Здесь спорно, так что оставь, как нравится, но, в целом, считается хорошей практикой делать такие название булевых переменных, которые сами содержат ответ на вопрос, на который можно ответить "да" или "нет". Например на вопрос "isDisabled" можно ответить "да" или "нет". 
Здесь спорно, конкретно, потому что `disabled` это дефолтный атрибут в html и здесь это может быть допустимо.

seal-developer replied:
Сделала, как лучше)

Tauders replied:
Здесь вкусовщина, а не совсем "однозначно лучше", так что может и не точно :)

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,31 @@
+type ButtonProps = {
+  className?: string;
+  onClick?: React.FormEventHandler<Element> | React.MouseEventHandler<Element>;
+  onKeyDown?: React.FormEventHandler<Element>;
```
А почему здесь `FormEventHandler`? Можно ли здесь указать более конкретный тип, а не `Element`? (например `HTMLButtonElement`), и почему бы не указать его?

seal-developer replied:
FormEventHandler нужен был бы если бы использовала отдельно onSubmit в форме, но раз через onClick это делаю, тогда не нужен, предполагаю.
Указала конкретный элемент.

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,31 @@
+  disabled,
+}: ButtonProps) => {
+  let classesNames = className
+    ? className + ' ' + classes.button
```
Вот это проще сделать через [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

seal-developer replied:
Использовала библиотеку

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -0,0 +1,34 @@
+};
+
+export const Card = ({ title, description, card, onDelete }: CardProps) => {
+  const deleteCard: React.MouseEventHandler = () => {
```
Здесь можно применить хорошую практику (конечно, не обязательно) и называть методы внутри компонентов с слов "handle", то есть будет `handleDeleteCard`. Это не обязательно, и о таком зачастую договариваются внутри команды.

seal-developer replied:
Поправила.

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -0,0 +1,34 @@
+  title: string;
+  description: string;
+  card: Task;
+  onDelete: Function;
```
А зачем здесь такой тип? Лучше будет `(card: Card) => void` или `(cardId: string) => void`

seal-developer replied:
Поправила.

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -0,0 +1,34 @@
+          children={
+            <path d="M17.8,2.5h-2.62A2.813,2.813,0,0,0,12.385,0H8.222A2.814,2.814,0,0,0,5.43,2.5H2.81A2.814,2.814,0,0,0,0,5.308H0A9.38,9.38,0,0,1,1.249,9.816v8.4a2.4,2.4,0,0,0,2.394,2.393H16.964a2.4,2.4,0,0,0,2.393-2.393v-8.4a9.383,9.383,0,0,1,1.248-4.508h0A2.813,2.813,0,0,0,17.8,2.5ZM8.222.625h4.163A2.188,2.188,0,0,1,14.548,2.5H6.058A2.19,2.19,0,0,1,8.222.625ZM18.733,9.816v8.4a1.771,1.771,0,0,1- ...
+          }
+        />
```
Попробуй использовать библиотеку vite-plugin-svgr - это позволит использовать свг напрямую, и хранить их в формате свг. Так будет проще их поддерживать и добавлять.

seal-developer replied:
Поменяла на библиотеку

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -0,0 +1,34 @@
+  const deleteCard: React.MouseEventHandler = () => {
+    onDelete(card);
+  };
+  return (
```
Вкусовщина, но перед return хорошо смотрится пустая строка :)

seal-developer replied:
Добавила пустую строку

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+  onAddCard,
+  closeModal,
+}: FormAddingCardProps) => {
+  const [title, setTitle] = useState<string>('');
```
Можно не уточнять тип в случае, если по дефолтному значению можны вычислить типа днных.

seal-developer replied:
Без типа ругается

Tauders replied:
А на каком этапе? Если я так делаю
```ts
  const [title, setTitle] = useState('');
```
у меня не ругается, так как есть дефолтное значение, из которого ТС может вычислить тип однозначно.

seal-developer replied:
Хм, возможно, я что-то тыкнула случайно, сейчас все ок

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+  const [description, setDescription] = useState<string>('');
+  const [isDisabled, setIsDisabled] = useState(true);
+
+  const handleChangeTitle: React.FormEventHandler<Element> = (
```
Почему такой тип у этого метода?

seal-developer replied:
Уже поняла, по похожему комментарию, что это не подходит сюда

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setTitle(e.currentTarget.value);
+    if (title !== '') setIsDisabled(false);
```
Лучше добавлять фигурные скобочки, иначе легко ошибиться

seal-developer replied:
Добавила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+    if (title !== '') setIsDisabled(false);
+  };
+
+  const handleChangeDescription: React.FormEventHandler<Element> = (
```
Почему такой тип у этого метода?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+    }
+  }, [title, description]);
+
+  const onSubmit: React.FormEventHandler<Element> = (e: React.FormEvent) => {
```
Почему `onSubmit`, а не `handleSubmit`? Почему событие не форме самой, а на кнопке?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+        onChange={handleChangeDescription}
+        placeholder="Description"
+      />
+      <Button disabled={isDisabled} children={'Add card'} onClick={onSubmit} />
```
Почему `children` так назначен?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+type FormAddingCardProps = {
+  className?: string;
+  cards: Array<Task>;
+  onAddCard: Function;
```
Это не очень хороший тип, лучше прописать конкрентее, например `onAddCard: (card: Task) => void`

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+
+    const newCards = [...cards, newCard];
+
+    onAddCard(newCards);
```
Почему эта логика здесь, а не просто возвращение новой карты?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+  className?: string;
+  cards: Array<Task>;
+  onAddCard: Function;
+  closeModal: Function;
```
Почему не `onCloseModal`?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,90 @@
+import { v4 as uuidv4 } from 'uuid';
+
+type FormAddingCardProps = {
+  className?: string;
```
Зачем здесь `className`, если он не используется?

seal-developer replied:
Условно, компонент, может быть в дальнейшем и переиспользован, и тогда дополнительный класс может пригодиться, поправила, но такой логикой руководствовалась

Tauders replied:
Есть такой хороший метод YAGNI, буквально "тебе это не понадобится". (You aren't gonna need it). Смысл такой, что если тебе сейчас это не надо, но, в будущем, возможно, понадобится, то в будущем и можно будет добавить. Если что-то не используется здесь и сейчас, нужно ли оно здесь и сейчас? 
Расширяемость компонентов (потенциальная) хорошо, но это требует дополнительной поддержки и дополнительной работы с кодом, что влечёт дополнительные усилия, которые в данный момент не особо нужны.

seal-developer replied:
Хорошо

Tauders commented on ReactTasks/task-1/stage-2/src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,48 @@
+import { Icon } from '../Icon/Icon';
+
+type ModalProps = {
+  className?: string;
```
Зачем здесь className?

seal-developer replied:
Ответила в другом комментарии

Tauders commented on ReactTasks/task-1/stage-2/src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,48 @@
+  title: string;
+  children: React.ReactNode;
+  isOpen: boolean;
+  closeModal: React.MouseEventHandler;
```
Почему не `onCloseModal`?

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/ts/classes/simpleCard/simpleCard.ts:
```diff
@@ -0,0 +1,19 @@
+  title: string;
+  description: string;
+}
+export default class simpleCard implements Card {
```
Зачем делать это в виде класса?

seal-developer replied:
Чтобы потом легче было расширять?

Tauders replied:
А как это упростит дальнейшее расширение? Ты планируешь использовать это не просто для передачи данных, а для чего-то ещё? На текущий момент всё, что угодно, что имеет интерфейс `Card` тебе подходит (особенности типизации ТС), так зачем делать класс, который содержит в себе только поля, и который будет не сильно полезен без методов и тд?

seal-developer replied:
Убрала

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,87 @@
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    setCards(
+      cards
+        .map((task, index) => {
+          task.id = uuidv4();
+          task.order = index + 1;
+          return task;
+        })
+        .sort(sortByCardOrder)
+    );
```
Не очень понимаю, а зачем здесь эта логика? Может проще при назначении дефолтных значений для `cards` это сделать?

seal-developer replied:
Эта логика нужна, так как мне надо искусственно созданные данные еще подгружать, и сортировать их

Tauders replied:
Извини, не понимаю, какие искусственно созданные данные? Почему, когда ты назначаешь дефолтное значение в useState, это не подходит?

Tauders replied:
Грубо говоря, почему бы не сделать так?
```
  const [cards, setCards] = useState<Task[]>(
    dataTasks
      .map((task, index) => {
        task.id = uuidv4();
        task.order = index + 1;
        return task;
      })
      .sort(sortByCardOrder)
  );
```
?

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,87 @@
+        {cards.map((item, index) => {
+          return (
+            <Card
+              key={index}
```
А почему бы не использовать id?

seal-developer replied:
Cтроковые id не нравятся, как ключи реакту
![image](https://github.com/Tauders/dratuIT/assets/87136385/976133bc-c2e9-4572-a360-84800732f863)
Но согласна, что индексы не безопасны, оставила только цифры для ключа

Tauders replied:
А у тебя все элементы имеют реально валидный id? В том плане, что строка то реакту нормально должна подходить, по типизации. Судя по типам
```
type Key = string | number | bigint;
```

Tauders replied:
Я понял, почему. У тебя при первом рендере нет `id` ещё, если поменяешь, чтобы useState дефолтные значения сразу ставил с id, такая проблема пропадёт, так как id будет всегда.

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,87 @@
+      </Modal>
+      <h2 className={classes.todoListSection__title}>{title}</h2>
+      <ul className={classes.todoListList}>
+        {cards.map((item, index) => {
```
Почему не назвать `card`, а не `item`?

seal-developer replied:
Поменяла

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -0,0 +1,34 @@
+import classes from './Card.module.scss';
+
+type CardProps = {
+  title: string;
+  description: string;
+  card: Task;
```
Зачем передевать `title`, `description` и `card`, откуда эти значения и берутся?

seal-developer replied:
title, description  из объектов массива с данными, card это один элемент массива, получаемый при использовании .map()

Tauders replied:
Не понимаю, у тебя есть такое использование.
```ts
              key={card.id}
              title={card.title}
              description={card.description}
              onDelete={handleDeleteCard}
              card={card}
```
Ты передаёшь card и данные же из него.
Почему бы не передавать только card и не рендерить внутри теже данные?
```ts
type CardProps = {
  card: Task;
  onDelete: (card: Task) => void;
};

export const Card = ({ card, onDelete }: CardProps) => {
  const handleDeleteCard: React.MouseEventHandler = () => {
    onDelete(card);
  };

  return (
    <li className={classes.card}>
      <h3 className={classes.card__title}>{card.title}</h3>
      <p className={classes.card__description}>{card.description}</p>
      <Button className={classes.card__button_light} onClick={handleDeleteCard}>
        <DeleteIcon
          width={16}
          height={16}
          className={classes.card__icon}
          preserveAspectRatio="xMidYMid meet"
          color="#7c4886"
        />
      </Button>
    </li>
  );
};
```
и оставить 
```ts
            <Card
              key={card.id}
              onDelete={handleDeleteCard}
              card={card}
            />
```

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,87 @@
+    setCards(cards.sort(sortByCardOrder));
+  };
+
+  const openModal = () => {
+    setIsOpen(true);
+  };
+  const closeModal = () => {
+    setIsOpen(false);
+  };
```
Можно назвать handle...

seal-developer replied:
Переименовала

Tauders reviewed (changes requested):
Есть бага, что если ты напечатаешь один символ в название, то он неправильно обновит статус выключено. Посмотри на useState и принцип его работы

seal-developer committed:
[task-1, stage 2] 1. Поправила отсутствие пустых строк;
2. Добавила библиотеки для классов и svg;
3. Поправила именование;
4. Поправила типы;
5. Поменяла реализацию ключей;
6. Поправила вызов события при отправке формы;

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -1,29 +1,27 @@
   children,
-  disabled,
+  isDisabled,
 }: ButtonProps) => {
```
Опять же, здесь вкусовщина, но есть другой способ так разбивать. 
```ts
export const Button = (props: ButtonProps) => {
  const { className, onClick, onKeyDown, children, isDisabled } = props;

  return (
    <button
      className={classNames(className && className, classes.button)}
      onClick={onClick}
      onKeyDown={onKeyDown}
      disabled={isDisabled}
    >
      {children}
    </button>
  );
};
```
Мне, лично, (оставляю на твой вкус), такое нравится больше.

seal-developer replied:
Изменила

Tauders commented on ReactTasks/task-1/stage-2/src/components/Button/Button.tsx:
```diff
@@ -1,29 +1,27 @@
   return (
     <button
-      className={classesNames}
+      className={classNames(className && className, classes.button)}
```
Можно по-другому использовать библиотеку, будет чуть понятнее выглядеть:
```ts
className={classNames(classes.button, { className: className })}
```
или даже так:
```ts
className={classNames(classes.button, { className: Boolean(className) })}
```

seal-developer replied:
На мой взгляд мой вариант понятнее, лично мне по крайней мере, если только в этом преимущество)
Может , что еще надо добавить, у меня оба твоих варианта не отрабатывают

Tauders replied:
Окей, тогда можно и так оставить

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -1,32 +1,31 @@
+        <DeleteIcon
           width={16}
           height={16}
+          className={classes.card__icon}
+          preserveAspectRatio="xMidYMid meet"
```
Оно так и должно выглядеть?
![image](https://github.com/Tauders/dratuIT/assets/4114636/77a4e4a0-4ed7-4276-96ab-250b868a2846)
Там и должен быть цвет фона поменян, а не цвет линий?

seal-developer replied:
У меня все ок - 
![image](https://github.com/Tauders/dratuIT/assets/87136385/f7dd9f7d-4518-4e85-8248-4b9f1fc7e19b)

Но если ты менял что-то с классами в кнопке, когда смотрел, могло из-за этого не отработать

Tauders replied:
Да, я сам накосячил :)

Tauders commented on ReactTasks/task-1/stage-2/src/components/Card/Card.tsx:
```diff
@@ -1,32 +1,31 @@
           width={16}
           height={16}
+          className={classes.card__icon}
+          preserveAspectRatio="xMidYMid meet"
```
Нужен ли аттрибут `preserveAspectRatio`? Может его лучше в саму svg переместить? Можно ли размер указать через класс?

seal-developer replied:
Поместила в svg, указала размеры через класс

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -6,34 +6,30 @@ import { Task } from '../../data/dataTasks';
   cards: Array<Task>;
-  onAddCard: Function;
-  closeModal: Function;
+  onAddCard: (cards: Task) => void;
```
Почему название `cards`, если это одна таска? Почему вообще `cards`, а не `task`?

seal-developer replied:
Забыла поправить, когда меняла после прошлых комментариев

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -6,34 +6,30 @@ import { Task } from '../../data/dataTasks';
     e: React.ChangeEvent<HTMLInputElement>
   ) => {
     setTitle(e.currentTarget.value);
-    if (title !== '') setIsDisabled(false);
```
Ты интересно поправила багу, но ты прочитала, почему бага возникала? Можешь объяснить?

seal-developer replied:
Ну потому что оно должно еще раз отрендериться, чтобы новое состояние установить, у него там еще старое на тот момент состояние, когда я проверку делаю

Tauders replied:
Да, это важно понимать, так как может вести к багам.

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -6,34 +6,30 @@ import { Task } from '../../data/dataTasks';
   const [isDisabled, setIsDisabled] = useState(true);
 
-  const handleChangeTitle: React.FormEventHandler<Element> = (
+  const handleChangeTitle: React.ChangeEventHandler<HTMLInputElement> = (
```
Вкусовщина, но можно убрать возвращаемый тип

seal-developer replied:
Убрала

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -45,9 +41,14 @@ export const FormAddingCard = ({
+    e: React.FormEvent<HTMLFormElement>
+  ) => {
     e.preventDefault();
     const prevCard: Task = cards[cards.length - 1];
```
Если мы подразумеваем некое переиспользование (что мы должны делать, в целом), может лучше возвращать без `order`, вычисленного здесь, а вычислять его в родительском компоненте? Так не надо будет передавать сюда все карточки, и сама форма добавления не будет контролировать то, в каком порядке что добавляется.

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/Modal/Modal.tsx:
```diff
@@ -1,43 +1,35 @@
   children: React.ReactNode;
   isOpen: boolean;
-  closeModal: React.MouseEventHandler;
+  onCloseModal: React.MouseEventHandler;
```
Вкусовщина, но важно ли нам, что это именно мышкой кликнутый ивент? В плане, а что если там будет ещё клик на клавиатуру? Не лучше ли будет сделать более общий вариант, типа `() => void`?

seal-developer replied:
Поменяла

Tauders commented on ReactTasks/task-1/stage-2/src/components/Modal/Modal.tsx:
```diff
@@ -1,43 +1,35 @@
+              preserveAspectRatio="xMidYMid meet"
               width={16}
               height={16}
               color="#7c4886"
```
Так же, как для кнопки Delete

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -34,52 +34,50 @@ export const TodoList = ({ title }: TodoListProps) => {
       >
-        <Icon
+        <AddIcon
+          className={classes.todoListSection__icon}
```
Так же, как для delete,

seal-developer replied:
Поправила

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -0,0 +1,85 @@
+
+  return (
+    <form className={classes.form} onSubmit={handleSubmit}>
+      <input
+        className={classes.form__input}
+        value={title}
+        onChange={handleChangeTitle}
+        type="text"
+        placeholder="Title"
+      />
+      <textarea
+        className={classes.form__textarea}
+        value={description}
+        onChange={handleChangeDescription}
+        placeholder="Description"
+      />
```
Может стоит заиспользовать дефолтные валидации от html для форм? Тогда кнопка всегда будет активна, но, пока пользователь не введёт данные, он не сможет отправить их. Так можно убрать useEffect.

seal-developer replied:
Поменяла

seal-developer committed:
[task 1, stage 2] Перенесла  preserveAspectRatio в svg файлы.

seal-developer committed:
[task 1, stage 2] Убрала размеры иконок в стили.

seal-developer committed:
[task 1, stage 2] Добавила стили для кнопки при HTML валидации полей.

seal-developer committed:
[task 1, stage 2] Убрала useEffect, перенесла добавление id и order, сортировку в useState для начального значения. Убрала лишнее из иконки. Убрала лишнее при формировании ключа у карточки.

seal-developer committed:
[task 1, stage 2] Убрала логику с классом для карточки, перенесла формирование order в родительский компонент. Добавила HTML валидацию, убрала UseEffect для блокирования кнопки.

seal-developer committed:
[task 1, stage 2]  Работа с типами, обновление компонентов иконок.
Убрала пропсы title и description, сделала передачу через card.

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -13,29 +13,36 @@ type TodoListProps = {
-    );
-  }, []);
+  const [cards, setCards] = useState<Task[]>(
+    dataTasks
+      .map((task, index) => {
+        task.id = uuidv4();
+        task.order = index + 1;
+        return task;
+      })
+      .sort(sortByCardOrder)
```
Если такое кажется не очень понятным и не очень хорошо смотрится, можно убрать это в подготовку данных. То есть сделать метод в в файле dataTasks, который будет это делать в файле.

Tauders commented on ReactTasks/task-1/stage-2/src/components/TodoList/TodoList.tsx:
```diff
@@ -13,29 +13,36 @@ type TodoListProps = {
+        ...cards,
+        {
+          ...newCard,
+          order: cards.length > 0 && prevCard.order ? prevCard.order + 1 : 1,
```
Здесь это не так важно, так как индекс от 1, но в целом проверять числа вот так может привести к проблеме, если там валидное значение 0. Так как такое вычисление при 0 будет неверным. (условие не выполнится)

Tauders commented on ReactTasks/task-1/stage-2/src/components/FormAddingCard/FormAddingCard.tsx:
```diff
@@ -37,26 +31,13 @@ export const FormAddingCard = ({
     e.preventDefault();
-    const prevCard: Task = cards[cards.length - 1];
-    const newCard = new SimpleCard({
+    const newCard = {
```
Здесь бы в идеале указать тип как раз, чтобы проверка на данные была, типа 
```
const newTask: Task = {
...
```

Tauders reviewed (changes requested):
Я бы ещё предложил подумать над организацией файлов, чтобы она была чуть проще для расширения дальнейшего.
Один из таких подходов, это делить всё на компоненты и страницы. Страницы это нечто, что можно отрендерить по финальному урлу.
Оно может выглядеть как-то так:
![image](https://github.com/Tauders/dratuIT/assets/4114636/47bb3561-dfee-4131-a393-d58b2eda3b8f)

seal-developer committed:
[task 1, stage 2] Перенесла подготовку массива в файл с данными

seal-developer committed:
[task 1, stage 2] Изменила проверку при добавлении order; указала тип при создании новой карточки

seal-developer committed:
[task 1, stage 2] Декомпозиция файлов

seal-developer committed:
[task 1, stage 2] Убрала излишний тип

seal-developer committed:
[task 1, stage 2] Изменила функцию prepareDataTasks
